backend/src/openarchiefbeheer/destruction/models.py

```diff
@@ -235,33 +235,38 @@ def generate_destruction_report(self) -> None:
             "zaaktype omschrijving",
             "selectielijst procestype nummer",
         ]
-        with NamedTemporaryFile(mode="w", newline="", delete_on_close=False) as f_tmp:
-            writer = csv.DictWriter(f_tmp, fieldnames=fieldnames)
-            writer.writeheader()
-            for item in self.items.filter(
-                processing_status=InternalStatus.succeeded
-            ).iterator(chunk_size=1000):
-                data = {
-                    **item.extra_zaak_data,
-                    **{
-                        "zaaktype url": item.extra_zaak_data["zaaktype"]["url"],
-                        "zaaktype omschrijving": item.extra_zaak_data["zaaktype"][
-                            "omschrijving"
-                        ],
-                        "selectielijst procestype nummer": item.extra_zaak_data[
-                            "zaaktype"
-                        ]["selectielijst_procestype"]["nummer"],
-                    },
-                }
-                del data["zaaktype"]
-
-                writer.writerow(data)
+        with NamedTemporaryFile(mode="wb", delete_on_close=False) as f_tmp:
+            workbook = xlsxwriter.Workbook(f_tmp.name, options={"in_memory": False})
+            worksheet = workbook.add_worksheet(name=gettext("Deleted zaken"))
+            worksheet.write_row(0, 0, fieldnames)
+
+            for row_count, item in enumerate(
+                self.items.filter(processing_status=InternalStatus.succeeded).iterator(
+                    chunk_size=1000
+                )
+            ):
+                data = [
+                    item.extra_zaak_data["url"],
+                    item.extra_zaak_data["einddatum"],
+                    item.extra_zaak_data["resultaat"],
+                    item.extra_zaak_data["startdatum"],
+                    item.extra_zaak_data["omschrijving"],
+                    item.extra_zaak_data["identificatie"],
+                    item.extra_zaak_data["zaaktype"]["url"],
+                    item.extra_zaak_data["zaaktype"]["omschrijving"],
+                    item.extra_zaak_data["zaaktype"]["selectielijst_procestype"][
+                        "nummer"
+                    ],
+                ]
+
+                worksheet.write_row(row_count + 1, 0, data)
+            workbook.close()
 
             f_tmp.close()
-            with open(f_tmp.name, mode="r") as f:
+            with open(f_tmp.name, mode="rb") as f:
```

Comment:
Is there a reason of closing and then re-opening the file (maybe the write/read handle)? Can't we close after the final save and use a single handle?

Reply:
Yes! The django `FieldFile` expects a `File` object. So we could either generate the content in memory with StringIO/BytesIO and then pass it directly to the `FieldFile`, or write to a temporary file and then pass that to the `FieldFile`. 
Since we could have many zaken, I worried that keeping it in memory would lead to out-of-memory errors so I went for the writing to file approach.
However, it seems that the temporary file cannot be opened in read/write mode at the same time, so you have to first close it and then open it again in read mode.

Maybe there is a better way of doing this, but I haven't found it yet :S

Reply:
I think that in the implementation, the resulting file is still completely loaded into memory because of the Django File logic, maybe we could play with the chucks parameter?

https://docs.djangoproject.com/en/5.1/ref/files/file/#django.core.files.File.chunks